sgkit/window.py

```diff
@@ -0,0 +1,206 @@
+from typing import Any, Callable, Tuple
+
+import dask.array as da
+import numpy as np
+from xarray import Dataset
+
+from sgkit.utils import conditional_merge_datasets
+
+from .typing import ArrayLike, DType
+
+# Window definition (user code)
+
+
+def window(
+    ds: Dataset,
+    size: int,
+    step: int,
+    merge: bool = True,
+) -> Dataset:
+    """Add fixed-size windowing information to a dataset.
+
+    Windows are defined over the ``variants`` dimension, and are
+    used by some downstream functions to calculate statistics for
+    each window.
+
+    Parameters
+    ----------
+    ds
+        Genotype call dataset.
+    size
```

Comment:
Hmm, I'm not so sure about this API, it feels a bit inflexible. People are going to want to window and plot in terms of physical distance and not just in terms of the number of variants (see the proposal in #229). I see that using the number of variants aligns much more nicely with chunking. Is this a fundamental restriction - will we always need windows defined in this way to have equal numbers of variants? If we did want to do things like "compute Fst in 100Kb windows", is this incompatible with the way we're doing things in Dask?

It's not obvious what the utility of the ``step`` parameter is either - I'd imagine the vast majority of people will want contiguous windows (unless I'm misunderstanding what this does?). It should at least have a default such that users do get contiguous windows and don't need to specify this value all the time.

Reply:
I really like this idea @tomwhite - how about we create a follow up issue for this so, and get this much merged?

Reply:
Filed #341